services/director-v2/src/simcore_service_director_v2/modules/comp_scheduler/factory.py

```diff
@@ -0,0 +1,71 @@
+import logging
+from typing import List, cast
+
+from fastapi import FastAPI
+
+from ...core.errors import ConfigurationError
+from ...models.domains.comp_runs import CompRunsAtDB
+from ...utils.scheduler import SCHEDULED_STATES, get_repository
+from ..db.repositories.comp_runs import CompRunsRepository
+from .base_scheduler import BaseCompScheduler, ScheduledPipelineParams
+
+logger = logging.getLogger(__name__)
+
+
+async def create_from_db(app: FastAPI) -> BaseCompScheduler:
+    if not hasattr(app.state, "engine"):
+        raise ConfigurationError(
+            "Database connection is missing. Please check application configuration."
+        )
+    db_engine = app.state.engine
+    runs_repository: CompRunsRepository = cast(
+        CompRunsRepository, get_repository(db_engine, CompRunsRepository)
+    )
+
+    # get currently scheduled runs
+    runs: List[CompRunsAtDB] = await runs_repository.list(
+        filter_by_state=SCHEDULED_STATES
+    )
+
+    logger.debug(
+        "Following scheduled comp_runs found still to be scheduled: %s",
+        runs if runs else "NONE",
+    )
+
+    # TODO: here we need to persist the pipeline stopping part
+
+    # check which scheduler to start
+    if (
+        app.state.settings.CELERY_SCHEDULER.DIRECTOR_V2_CELERY_SCHEDULER_ENABLED
+        and not app.state.settings.DIRECTOR_V2_DEV_FEATURES_ENABLED
+    ):
+        logger.info("Creating Celery-based scheduler...")
+        from ..celery import CeleryClient
```

Comment:
Is there any reason for embedded imports? they are both there, right?

Reply:
no reason. fixed.